### DISCUSSION TITLE
詠唱魔法 Stable Diffusion AI 算圖

### DISCUSSION THREAD

**Discussion by ThanatosJun** [1101-GS4719-C/Python-Community-GS4719-C#1156] [General]
最近又再度迷上了AI算圖，而此次使用的AI算圖工具為Stable Diffusion。相較於最初使用的Disco Diffusion，它運算的速度更快，圖像也更精緻，在效能上 Stable Diffusion 可以說是 Disco Diffusion 的次方級上位版。如果問我為何有段時間停止AI算圖這麼有趣的事？其中一個原因便是算圖次數的問題！！與Dell‧E2 和 Midjourney 這兩項我使用過的工具相比，Stable Diffusion 不僅是免費的，而且還沒有運算次數限制的存在，唯一一個需要解決的問題就是你的GPU給不給力！不過有個辦法可以解決這個問題，使用 Google Cloab 吧~

使用 Stable Diffusion webui 有兩個方式，第一個方式需要有強大的獨顯，本機端下載並用本機端的GPU運算，雖然比較方便，但可憐如我，4GB 的 RTX 3050 Ti 雖然能夠開啟使用者介面，但運算開始時 VRAM 完全不夠用，導致跑圖跑到一半就斷了...... 至少 8GB 的N卡獨顯才夠用。第二個方式就親民的多了，我們只需要一個Google 雲端帳號就可以詠唱咒語，召喚各種美圖、怪圖、奇葩圖。雖然免費的線上GPU很香，但是免費的有使用額度限制，不過一天三、四個小時已經足夠用了！原本我想用學校免費且不限制的雲端硬碟來儲存各種模型與圖片，但最近教育資源只剩100TB的儲存內容，我只得放棄，改用我15GB的私人帳號來進行詠唱了！因為一個模型的內容頗大，所以雲端內無法同時加載好幾個模型來切換使用，只能固定幾個模型，以舊換新，不然升級容量也是個辦法！！

目前我知悉的 Stable Diffusion 基礎生成方式為：
1. 套用模型 
2. 輸入文字提詞與初始化噪聲圖像 
3. 放入UNET網絡 
4. Decoder 解碼 還原清晰圖像 
5. image out

Stable Diffusion 有著許多輔助的模型、結構與神經網路，如最近的Lora和異軍突起的ControlNet，有好多好多值得被我們研究與發現，甚至是發明！！不過對於我來說，現階段還是研究各位大神的開發，並多多了解 Stable Diffusion 的運作機制與操作介面為好！！
歡迎大家多多指教，並加入詠唱師的行列~

以下分享一些近日生成的圖片
![00091-1239576930](https://user-images.githubusercontent.com/122719731/225584850-da27205a-22fd-4161-b92d-390818528cd8.png)
![00196-2823325734](https://user-images.githubusercontent.com/122719731/225584893-c87ce4ec-c223-4e02-9450-58c1d475d0c6.png)

**Comment by JohnnyXiao1024** [comment]
感謝同學分享AI算圖的主題
近年隨著AI繪圖的興起，利用精準的題詞產生優秀的畫作已然成為一們新顯學
分享兩個好用的網站給大家：[Prompt Hero](https://prompthero.com/) 和 [clip interrogator](https://replicate.com/pharmapsychotic/clip-interrogator)

Prompt Hero 上搜集了許多aI生成的圖片，並提供詠唱過程，供大家參考
而如果你手上有一張AI圖片卻不知道它的提詞，則可以使用 clip interrogator 這項工具
上傳後幾分鐘，網站便會列出這張圖可能的提詞，以下是我利用樓主作品做的示範：

<img width="1400" alt="截圖 2023-03-21 上午12 29 11" src="https://user-images.githubusercontent.com/125842539/226406008-7c0f1576-7dae-4c58-82ca-bb3f8bf61153.png">

希望這些工具可以協助大家提升詠唱技能～
